File: portfolio-react/src/components/WorkExperience.js
import React from 'react';
import './WorkExperience.css';

function WorkExperience({ workHistory }) {
  return (
    <section className="section work-experience">
      <h2 className="section-title">Work Experience</h2>
      
      {workHistory.map((job, index) => (
        <article key={index} className="job">
          
          <div className="job-header">
            <h3 className="company">{job.company}</h3>
            <span className="date-range">{job.dateRange}</span>
          </div>
          
          <p className="position">{job.position}</p>
          <p className="location">{job.location}</p>
          
          {/* what I did at this job */}
          <ul className="responsibilities">
            {job.responsibilities.map((resp, idx) => (
              <li key={idx}>{resp}</li>
            ))}
          </ul>

        </article>
      ))}
    </section>
  );
}

export default WorkExperience;

File: src/model/WorkExperience.java
package model;

import java.util.List;

public class WorkExperience {
    private String company;
    private String location;
    private String position;
    private String dateRange;
    private List<String> responsibilities;

    public WorkExperience(String company, String location, String position, String dateRange, List<String> responsibilities) {
        this.company = company;
        this.location = location;
        this.position = position;
        this.dateRange = dateRange;
        this.responsibilities = responsibilities;
    }

    public String getCompany() { return company; }
    public String getLocation() { return location; }
    public String getPosition() { return position; }
    public String getDateRange() { return dateRange; }
    public List<String> getResponsibilities() { return responsibilities; }
}
